posthog/api/survey.py

```diff
@@ -751,6 +773,249 @@ def responses_count(self, request: request.Request, **kwargs):
 
         return Response(counts)
 
+    def _validate_and_parse_dates(
+        self, date_from: str | None, date_to: str | None
+    ) -> tuple[datetime | None, datetime | None]:
+        """Validate and parse date_from and date_to.
+
+        Args:
+            date_from: Optional ISO timestamp for start date with timezone info
+            date_to: Optional ISO timestamp for end date with timezone info
+
+        Returns:
+            Tuple of (parsed_date_from, parsed_date_to) in UTC
+
+        Raises:
+            ValidationError: If dates are invalid or if date_from is after date_to
+        """
+        parsed_from = None
+        parsed_to = None
+
+        try:
+            if date_from:
+                parsed_from = datetime.fromisoformat(date_from).astimezone(UTC)
+
+            if date_to:
+                parsed_to = datetime.fromisoformat(date_to).astimezone(UTC)
+
+            if parsed_from and parsed_to and parsed_from > parsed_to:
+                raise exceptions.ValidationError("date_from must be before date_to")
+
+            return parsed_from, parsed_to
+
+        except ValueError:
+            raise exceptions.ValidationError(
+                "Invalid date format. Please use ISO 8601 format with timezone info (e.g. 2024-01-01T00:00:00Z or 2024-01-01T00:00:00+00:00)"
+            )
+
+    def _process_survey_results(self, results) -> SurveyStats:
+        """Process raw survey event results into stats format.
+
+        Args:
+            results: Raw results from ClickHouse query containing event stats
+
+        Returns:
+            Dictionary containing processed stats for each event type
+        """
+        # Initialize stats with zero values for all event types
+        stats: SurveyStats = {
```

Comment:
created new types so the linter catches if we're using non-existing properties @marandaneto